src/EditorFeatures/CSharpTest/CodeActions/ConvertLinq/ConvertForEachToLinqQueryTests.cs

```diff
@@ -0,0 +1,2842 @@
+                    if (x + y / 10 + z / 100 < 6)
+                    {
+                        r1.Add(g);
+                    }
+                }
+            }
+        }|]
+
+        Console.WriteLine(r1);
+    }
+}
+";
+            string output = @"
+using System;
+using System.Linq;
+using C = System.Collections.Generic.List<int>;
+class Query
+{
+    public static void Main(string[] args)
+    {
+        C c1 = new C { 1, 2, 3 };
+        C c2 = new C { 10, 20, 30 };
+        C c3 = new C { 100, 200, 300 };
+        C r1 = (from int x in c1
+                from int y in c2
+                from int z in c3
+                let g = x + y + z
+                where x + y / 10 + z / 100 < 6
+                select g).ToList();
+        Console.WriteLine(r1);
+    }
+}
+";
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForEachToQuery)]
+        public async Task ToListTypeReplacement02()
+        {
+            string source = @"
+using System.Linq;
+using System;
+using C = System.Collections.Generic.List<int>;
+class Query
+{
+    public static void Main(string[] args)
+    {
+        C c1 = new C { 1, 2, 3 };
+        C c2 = new C { 10, 20, 30 };
+        C r1 = new C();
+        [|foreach (int x in c1)
+        {
+            foreach (var y in c2)
+            {
+                if (Equals(x, y / 10))
+                {
+                    var z = x + y;
+                    r1.Add(z);
+                }
+            }
+        }|]
+
+        Console.WriteLine(r1);
+    }
+}
+";
+            string output = @"
+using System.Linq;
+using System;
+using C = System.Collections.Generic.List<int>;
+class Query
+{
+    public static void Main(string[] args)
+    {
+        C c1 = new C { 1, 2, 3 };
+        C c2 = new C { 10, 20, 30 };
+        C r1 = (from int x in c1
+                from y in c2
+                where Equals(x, y / 10)
+                let z = x + y
+                select z).ToList();
+        Console.WriteLine(r1);
+    }
+}
+";
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForEachToQuery)]
+        public async Task ToListPropertyAssignment()
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+public class Test
+{
+    public static void Main()
+    {
+        var nums = new int[] { 1, 2, 3, 4 };
+        var c = new C();
+        c.A = new List<int>();
+        [|foreach (var x in nums)
+        {
+            c.A.Add(x + 1);
+        }|]
+    }
+
+    class C
+    {
+        public List<int> A { get; set; }
+    }
+}";
+            string output = @"
+using System.Collections.Generic;
+using System.Linq;
+public class Test
+{
+    public static void Main()
+    {
+        var nums = new int[] { 1, 2, 3, 4 };
+        var c = new C();
+        c.A = (from x in nums
```

Comment:
does it check whether property has setter? or just do it and let squiggle to let users know the issue?

Reply:
I am +1 on just do it and let users know through squiggle. or warning in the preview.

Reply:
ah. sorry. it behave differently. awesome.